### PR TITLE
ninvaders: init at 0.1.1

### PR DESCRIPTION
**Pull request by zutesuit**
ninvaders: init at 0.1.1

This adds `ninvaders`, a Space Invaders clone for NixOS, based on the work from https://github.com/TheZ3ro/ninvaders.

Original game website: https://ninvaders.sourceforge.net

### Testing
- Built successfully using `nix-build`.
- Verified that the game runs as expected.

### CODE REVIEW

**Comment by zutesuit** [comment]
whitespace trails removed and .nix files formatted

File: pkgs/games/ninvaders/master.tar.gz
Comment (pluiedev) [inline pkgs/games/ninvaders/master.tar.gz:1]: Why is this here? We don't bundle source tarballs in Nixpkgs

File: pkgs/games/ninvaders/ninvaders.nix [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "ninvaders";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "TheZ3ro";
+    repo = "ninvaders";
+    rev = "master";
```
Comment (pluiedev) [inline pkgs/games/ninvaders/ninvaders.nix:15]: `rev` must always be a pinned commit (with the commit hash) or a tag

**Review by pluiedev — CHANGES_REQUESTED** [review]
Welcome to Nixpkgs! There's... a few problems.

**Comment by pluiedev** [comment]
Also, it seems to me that this is abandonware as upstream has been archived for literally more than 5 years. Generally we need more evidence of a need for unmaintained packages for inclusion in Nixpkgs since they can burden maintenance in the future

**Comment by pluiedev** [comment]
Wait — I just realized that this already exists in Nixpkgs and is at a newer version (0.1.2). Is this PR still relevant then?

**Comment by zutesuit** [comment]
Updated installPhase:
Replaced mkdir and cp commands with the install -Dm755 command for better adherence to Nixpkgs standards.

Included runHook preInstall and runHook postInstall.

Pinned the rev in fetchFromGitHub:
Updated rev to the latest commit hash (c6ab411) for reproducibility.


Restructured to the by-name system:
Moved ninvaders.nix to pkgs/by-name/ni/ninvaders/package.nix as per the new directory structure.

Removed the old pkgs/games/ninvaders directory and associated files.

Updated pkgs/top-level/default.nix, removing the game section.

**Comment by zutesuit** [comment]
> Wait — I just realized that this already exists in Nixpkgs and is at a newer version (0.1.2). Is this PR still relevant then?

well that's embarrasing. don't know why it didn't show up for me earlier. Sorry for the time waste, I'll cancel the PR then and try and find something useful and definitely not on the package list.
